### PR TITLE
Problem: Android build environment variables need clarifications.

### PR DESCRIPTION
**Pull request by stephan57160**
Reason: All are spread and initialized throughout the code.

Solution: Declare, initialize and document environment variables on top of build.sh.

Benefit: This participates to documentation.

### CODE REVIEW

**Comment by stephan57160** [comment]
4 or 5 more COMMIT still in the pipe, trying to bring some enhancements with no breaking changes in ANDROID CI build tools.

I keep last one to update the related Android CI build README.

File: builds/android/ci_build.sh [hunk 1]
```diff
@@ -3,12 +3,15 @@
 #   Exit if any step fails
 set -e
 
-export NDK_VERSION=android-ndk-r25
-export ANDROID_NDK_ROOT="/tmp/${NDK_VERSION}"
+# Configuration
+export NDK_VERSION="${NDK_VERSION:-android-ndk-r25}"
+export ANDROID_NDK_ROOT="${ANDROID_NDK_ROOT:-/tmp/${NDK_VERSION}}"
+export MIN_SDK_VERSION=${MIN_SDK_VERSION:-21}
+export ANDROID_BUILD_DIR="${ANDROID_BUILD_DIR:-$(cd $(dirname "${BASH_SOURCE[0]}") ; pwd)}"
```
[the remark this answers is not in the document]
Comment (bluca) [reply builds/android/ci_build.sh:10]: @stephan57160 some warnings to fix

**Comment by stephan57160** [comment]
Some more fixed, in existing code, with last commit
(only Android CI build tools are checked, modified & fixed).

**Comment by stephan57160** [comment]
On my side, there are no more reported warnings.
I just hope both shell checker won't conflict :-)

**Pull request merged**